{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "dcdeb2ea",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Import necessary libraries\n",
    "import yaml\n",
    "import pandas as pd\n",
    "\n",
    "# Load the CSV file\n",
    "csv_file_path = \"data/MethodList.csv\"  # Replace with the actual file name\n",
    "df = pd.read_csv(csv_file_path)\n",
    "df = df.loc[:, ~df.columns.str.contains('^Unnamed')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "555d2522",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Method</th>\n",
       "      <th>Publication</th>\n",
       "      <th>Year</th>\n",
       "      <th>Description</th>\n",
       "      <th>Model</th>\n",
       "      <th>Task</th>\n",
       "      <th>Inspired by</th>\n",
       "      <th>Code Availability</th>\n",
       "      <th>Published</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>cPCA</td>\n",
       "      <td>https://www.nature.com/articles/s41467-018-046...</td>\n",
       "      <td>2018</td>\n",
       "      <td>A modified version of PCA, where the covarianc...</td>\n",
       "      <td>Modified PCA</td>\n",
       "      <td>Linear Gene Programmes;Contrastive Disentangle...</td>\n",
       "      <td>PCA;Contrastive Mixture Models (Zou et al., 2013)</td>\n",
       "      <td>https://github.com/abidlabs/contrastive</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>CSMF</td>\n",
       "      <td>https://academic.oup.com/nar/article/47/13/660...</td>\n",
       "      <td>2019</td>\n",
       "      <td>A non-negative matrix factorisation that decom...</td>\n",
       "      <td>NMF</td>\n",
       "      <td>Linear Gene Programmes;Contrastive Disentangle...</td>\n",
       "      <td>iNMF;NMF</td>\n",
       "      <td>https://www.zhanglab-amss.org/homepage/softwar...</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>cLVM</td>\n",
       "      <td>https://arxiv.org/abs/1811.06094</td>\n",
       "      <td>2019</td>\n",
       "      <td>A family of contrastive latent variable models...</td>\n",
       "      <td>Factor Models;Contastive VAE</td>\n",
       "      <td>Linear Gene Programmes;Contrastive Disentangle...</td>\n",
       "      <td>Contrastive PCA</td>\n",
       "      <td>https://github.com/kseverso/contrastive-LVM</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>cVAE</td>\n",
       "      <td>https://arxiv.org/pdf/1902.04601</td>\n",
       "      <td>2019</td>\n",
       "      <td>VAE with two sets of latent variables (two enc...</td>\n",
       "      <td>Contrastive VAE</td>\n",
       "      <td>Contrastive Disentanglement</td>\n",
       "      <td>Contrastive PCA</td>\n",
       "      <td>https://github.com/abidlabs/contrastive_vae</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>scPCA</td>\n",
       "      <td>https://academic.oup.com/bioinformatics/articl...</td>\n",
       "      <td>2020</td>\n",
       "      <td>A sparse version of contrastive PCA that enhan...</td>\n",
       "      <td>Modified PCA</td>\n",
       "      <td>Linear Gene Programmes;Contrastive Disentangle...</td>\n",
       "      <td>Contrastive PCA;Probabilistic PCA</td>\n",
       "      <td>https://github.com/PhilBoileau/EHDBDscPCA</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>135</th>\n",
       "      <td>scPRAM</td>\n",
       "      <td>https://academic.oup.com/bioinformatics/articl...</td>\n",
       "      <td>2024</td>\n",
       "      <td>scPRAM is a computational framework for predic...</td>\n",
       "      <td>VAE;OT;Attention</td>\n",
       "      <td>Context Transfer;Trace Cell Populations</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/jiang-q19/scPRAM</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>136</th>\n",
       "      <td>MISTy</td>\n",
       "      <td>https://link.springer.com/article/10.1186/s130...</td>\n",
       "      <td>2022</td>\n",
       "      <td>MISTy extracts intra- and intercellular relati...</td>\n",
       "      <td>Spatially-informed;Random Forrest (or other re...</td>\n",
       "      <td>Feature Relationships</td>\n",
       "      <td>SVCA</td>\n",
       "      <td>https://saezlab.github.io/mistyR/</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>137</th>\n",
       "      <td>SpaCeNet</td>\n",
       "      <td>https://genome.cshlp.org/content/34/9/1371</td>\n",
       "      <td>2024</td>\n",
       "      <td>SpaCeNet aims to untangle the complex relation...</td>\n",
       "      <td>Generalised Gaussian Graphical Model;Spatially...</td>\n",
       "      <td>Feature Relationships</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/sschrod/SpaCeNet</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>Kasumi</td>\n",
       "      <td>https://www.nature.com/articles/s41467-025-594...</td>\n",
       "      <td>2025</td>\n",
       "      <td>Kasumi extends MISTy by focusing on identifyin...</td>\n",
       "      <td>Spatially-informed;Random Forrest (or other re...</td>\n",
       "      <td>Feature Relationships</td>\n",
       "      <td>MISTy</td>\n",
       "      <td>https://www.github.com/jtanevski/kasumi</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>Prescient</td>\n",
       "      <td>https://www.nature.com/articles/s41467-021-235...</td>\n",
       "      <td>2021</td>\n",
       "      <td>PRESCIENT models cellular differentiation as a...</td>\n",
       "      <td>Diffusion</td>\n",
       "      <td>Trace Cell Populations;Seen Perturbation Predi...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>https://github.com/gifford-lab/prescient</td>\n",
       "      <td>True</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>140 rows Ã— 9 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        Method                                        Publication  Year  \\\n",
       "0         cPCA  https://www.nature.com/articles/s41467-018-046...  2018   \n",
       "1         CSMF  https://academic.oup.com/nar/article/47/13/660...  2019   \n",
       "2         cLVM                   https://arxiv.org/abs/1811.06094  2019   \n",
       "3         cVAE                   https://arxiv.org/pdf/1902.04601  2019   \n",
       "4        scPCA  https://academic.oup.com/bioinformatics/articl...  2020   \n",
       "..         ...                                                ...   ...   \n",
       "135     scPRAM  https://academic.oup.com/bioinformatics/articl...  2024   \n",
       "136      MISTy  https://link.springer.com/article/10.1186/s130...  2022   \n",
       "137   SpaCeNet         https://genome.cshlp.org/content/34/9/1371  2024   \n",
       "138     Kasumi  https://www.nature.com/articles/s41467-025-594...  2025   \n",
       "139  Prescient  https://www.nature.com/articles/s41467-021-235...  2021   \n",
       "\n",
       "                                           Description  \\\n",
       "0    A modified version of PCA, where the covarianc...   \n",
       "1    A non-negative matrix factorisation that decom...   \n",
       "2    A family of contrastive latent variable models...   \n",
       "3    VAE with two sets of latent variables (two enc...   \n",
       "4    A sparse version of contrastive PCA that enhan...   \n",
       "..                                                 ...   \n",
       "135  scPRAM is a computational framework for predic...   \n",
       "136  MISTy extracts intra- and intercellular relati...   \n",
       "137  SpaCeNet aims to untangle the complex relation...   \n",
       "138  Kasumi extends MISTy by focusing on identifyin...   \n",
       "139  PRESCIENT models cellular differentiation as a...   \n",
       "\n",
       "                                                 Model  \\\n",
       "0                                         Modified PCA   \n",
       "1                                                  NMF   \n",
       "2                         Factor Models;Contastive VAE   \n",
       "3                                      Contrastive VAE   \n",
       "4                                         Modified PCA   \n",
       "..                                                 ...   \n",
       "135                                   VAE;OT;Attention   \n",
       "136  Spatially-informed;Random Forrest (or other re...   \n",
       "137  Generalised Gaussian Graphical Model;Spatially...   \n",
       "138  Spatially-informed;Random Forrest (or other re...   \n",
       "139                                          Diffusion   \n",
       "\n",
       "                                                  Task  \\\n",
       "0    Linear Gene Programmes;Contrastive Disentangle...   \n",
       "1    Linear Gene Programmes;Contrastive Disentangle...   \n",
       "2    Linear Gene Programmes;Contrastive Disentangle...   \n",
       "3                          Contrastive Disentanglement   \n",
       "4    Linear Gene Programmes;Contrastive Disentangle...   \n",
       "..                                                 ...   \n",
       "135            Context Transfer;Trace Cell Populations   \n",
       "136                              Feature Relationships   \n",
       "137                              Feature Relationships   \n",
       "138                              Feature Relationships   \n",
       "139  Trace Cell Populations;Seen Perturbation Predi...   \n",
       "\n",
       "                                           Inspired by  \\\n",
       "0    PCA;Contrastive Mixture Models (Zou et al., 2013)   \n",
       "1                                             iNMF;NMF   \n",
       "2                                      Contrastive PCA   \n",
       "3                                      Contrastive PCA   \n",
       "4                    Contrastive PCA;Probabilistic PCA   \n",
       "..                                                 ...   \n",
       "135                                                NaN   \n",
       "136                                               SVCA   \n",
       "137                                                NaN   \n",
       "138                                              MISTy   \n",
       "139                                                NaN   \n",
       "\n",
       "                                     Code Availability  Published  \n",
       "0              https://github.com/abidlabs/contrastive       True  \n",
       "1    https://www.zhanglab-amss.org/homepage/softwar...       True  \n",
       "2          https://github.com/kseverso/contrastive-LVM       True  \n",
       "3          https://github.com/abidlabs/contrastive_vae      False  \n",
       "4            https://github.com/PhilBoileau/EHDBDscPCA       True  \n",
       "..                                                 ...        ...  \n",
       "135                https://github.com/jiang-q19/scPRAM       True  \n",
       "136                  https://saezlab.github.io/mistyR/       True  \n",
       "137                https://github.com/sschrod/SpaCeNet       True  \n",
       "138            https://www.github.com/jtanevski/kasumi       True  \n",
       "139           https://github.com/gifford-lab/prescient       True  \n",
       "\n",
       "[140 rows x 9 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.head(140)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "31c6218c",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.head(140) # NOTE"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "ac218dfe",
   "metadata": {},
   "outputs": [],
   "source": [
    "df = df.fillna('-')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "a6112422",
   "metadata": {},
   "outputs": [],
   "source": [
    "df['Task'] = df['Task'].str.split(';')\n",
    "df['Model'] = df['Model'].str.split(';')\n",
    "df['Inspired by'] = df['Inspired by'].str.split(';')\n",
    "# remove first element of each list in 'Task' column\n",
    "df['Task'] = df['Task'].apply(lambda x: x[1:] if isinstance(x, list) and len(x[0])==0 else x)\n",
    "# df['Task'] = df['Task'].astype(str)\n",
    "df['Year'] = df['Year'].astype(int)\n",
    "\n",
    "# replace \\n with space\n",
    "df['Description'] = df['Description'].apply(lambda x: x.replace('\\n',' '))\n",
    "# Convert the DataFrame to a dictionary\n",
    "data_dict = df.to_dict(orient='records')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "637aa92a",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Save the dictionary as a YAML file\n",
    "yaml_file_path = \"methods.yaml\"  # Replace with the desired output file name\n",
    "with open(yaml_file_path, 'w') as yaml_file:\n",
    "    yaml.dump(data_dict, yaml_file, default_flow_style=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9873825e",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "37dd149b",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6ef9885c",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "cfe055be",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "dc3a6ccb",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "perspective",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.10"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
